### PR TITLE
Store language in localStorage

### PR DESCRIPTION
**Pull request by katevoss**
closes https://github.com/catalpainternational/canoe/issues/103

This PR does a few things:
1. When a user first loads the app, it looks for a `userLangage` in localStorage, which is where it is stored from their last visit. If they are a first-time user, then localStorage doesn't have the key.
2. Next it looks into their browser language. If the language is some variation of `en`, canoe changes the language to English.
3. Finally, if the browser's language is anything but English it sets the default language to Tetum.
4. When a user changes their language on the Settings page, it updates the language in Store and it stores the language in localStorage, so that it will persist in future sessions.

Feedback and suggestions welcome.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -6,7 +6,16 @@ import App from "RiotTags/App.riot.html";
 import { store, changeLanguage } from "ReduxImpl/Store";
 import "./scss/canoe.scss";
 
-changeLanguage('tet');
```
Comment (sandroroux) [inline src/app.js:9]: The site language is a Redux value. There's a more complicated, but more architectural way to do this: we should [initialize the Redux state](https://redux.js.org/recipes/structuring-reducers/initializing-state) with the correct language. Your code can set the Redux store's default state.

Wrap the code in a function:
```js
const getInitialLanguage = () => {
    const storedLanguage = localStorage.getItem(LANGUAGE_STORAGE_KEY);
    if (!!storedLanguage) {
        return storedLanguage;
    } else if (navigator.language.includes("en")) {
        return "en";
    } else {
        return "tet";
    }
}
```

and pass it to `createStore` as the `language` initial value:
```js
// This method is at Store.js's line 16.
const initialStoreState = {
    language: getInitialLanguage()
};

export const store = createStore(
    reducers,
    initialStoreState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
```

Let me know if you'd like me to explain this approach more.

File: src/js/redux/Store.js [hunk 2]
```diff
@@ -48,6 +48,9 @@ export const storeSiteDownloadedIs = (trueOrFalse) => {
 
 export const changeLanguage = (language) => {
     store.dispatch({ type: LANGUAGE_CHANGE, language: language });
+
+    const LANGUAGE_STORAGE_KEY = "userLanguage";
```
Comment (sandroroux) [inline src/js/redux/Store.js:52]: I'd suggest you define `LANGUAGE_STORAGE_KEY` once. If future developers must change the key, we want them to update one piece of code.

If you follow my other comment's approach, the key can stay in `Store.js`.

File: src/js/redux/Store.js [hunk 3]
```diff
@@ -48,6 +69,8 @@ export const storeSiteDownloadedIs = (trueOrFalse) => {
 
 export const changeLanguage = (language) => {
     store.dispatch({ type: LANGUAGE_CHANGE, language: language });
+
+    localStorage.setItem(LANGUAGE_STORAGE_KEY, language);
```
Comment (PeteCoward) [inline src/js/redux/Store.js:73]: I'd put this in the storeDispatch method inside Tranlsations.js

Comment (PeteCoward) [reply src/js/redux/Store.js:73]: or the setLocale method

Comment (PeteCoward) [reply src/js/redux/Store.js:73]: but I can see why it is here I suppose

Comment (sandroroux) [reply src/js/redux/Store.js:73]: Yeah, @PeteCoward. I'd rather hide the store behind this interface. In fact, we could move these read/write methods into their own file.

Comment (katevoss) [reply src/js/redux/Store.js:73]: @sandroroux I'm planning to merge this PR, did you want to me to move the read/write methods? Or should that be a separate issue?

Comment (sandroroux) [reply src/js/redux/Store.js:73]: Definitely a separate task. Thanks for merging.

**Pull request merged**